File: backend/app/models/cash_register.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Float, DateTime, JSON, Boolean
from ..database import Base


class CashRegister(Base):
    __tablename__ = "cash_registers"

    id = Column(String, primary_key=True, default=lambda: f"CAJ-{uuid.uuid4().hex[:6].upper()}")
    user_id = Column(String, nullable=False, index=True)
    open_date = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    close_date = Column(DateTime, nullable=True)
    initial_balance = Column(Float, default=0)
    current_balance = Column(Float, default=0)
    total_income = Column(Float, default=0)
    total_expense = Column(Float, default=0)
    status = Column(String(20), default="open")
    currency = Column(String(10), default="DOP")
    transactions = Column(JSON, default=list)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

File: dotnet-backend/src/Javaline.Commercial.Domain/Entities/CashRegister.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Javaline.Commercial.Domain.Common;

namespace Javaline.Commercial.Domain.Entities;

[Table("cash_registers")]
public class CashRegister : BaseEntity
{
    [Key]
    [Column("id")]
    public new string Id { get; set; } = string.Empty;

    [Column("user_id")]
    public string UserId { get; set; } = string.Empty;

    [Column("open_date")]
    public DateTime OpenDate { get; set; }

    [Column("close_date")]
    public DateTime? CloseDate { get; set; }

    [Column("initial_balance")]
    public decimal InitialBalance { get; set; }

    [Column("current_balance")]
    public decimal CurrentBalance { get; set; }

    [Column("total_income")]
    public decimal TotalIncome { get; set; }

    [Column("total_expense")]
    public decimal TotalExpense { get; set; }

    [Column("status")]
    public string Status { get; set; } = "open";

    [Column("currency")]
    public string Currency { get; set; } = "DOP";

    [Column("transactions")]
    public string Transactions { get; set; } = "[]";
}
